server.go

```diff
@@ -3781,6 +3979,9 @@ func newServer(ctx context.Context, listenAddrs []string, db database.DB,
 	if !cfg.SimNet && !cfg.RegNet && len(cfg.ConnectPeers) == 0 {
 		newAddressFunc = func() (net.Addr, error) {
 			for tries := 0; tries < 100; tries++ {
+				// Note that this does not filter by address type.  Unsupported
+				// network address types should be pruned from the address
+				// manager's internal storage prior to calling this function.
```

Comment:
`addrmgr.GetAddress` should accept a filter function argument that excludes network addresses that cannot be connected to.  For example, if dcrd is not routing traffic through a tor proxy then it should not attempt to fetch + connect to a TORv3 address stored in the address manager.